src/compiler/sys.ts

```diff
@@ -1064,13 +1057,39 @@ namespace ts {
             }
         }
 
+        function pollingWatchFile(fileName: string, callback: FileWatcherCallback, pollingInterval: PollingInterval, options: WatchOptions | undefined) {
+            return createSingleWatcherPerName(
+                pollingWatches,
+                useCaseSensitiveFileNames,
+                fileName,
+                callback,
+                cb => pollingWatchFileWorker(fileName, cb, pollingInterval, options),
+            );
+        }
         function fsWatch(
             fileOrDirectory: string,
             entryKind: FileSystemEntryKind,
             callback: FsWatchCallback,
             recursive: boolean,
             fallbackPollingInterval: PollingInterval,
             fallbackOptions: WatchOptions | undefined
+        ): FileWatcher {
+            return createSingleWatcherPerName(
+                recursive ? fsWatchesRecursive : fsWatches,
+                useCaseSensitiveFileNames,
+                fileOrDirectory,
+                callback,
+                cb => fsWatchHandlingPresence(fileOrDirectory, entryKind, cb, recursive, fallbackPollingInterval, fallbackOptions),
+            );
+        }
+
+        function fsWatchHandlingPresence(

```

Comment:
What does "handling presence" mean?

Reply:
Its watcher that handles if file or directory is present on disk and based on that either create file system event based watcher(if exists) or polling watcher (for missing)

Reply:
Renamed it to make it clearer